### PR TITLE
Fixing some connection status issues

### PR DESCRIPTION
**Pull request by CliftonH**
This PR does a few different things. First, the status announcements in `web/index.js` in response to the `'online'` and `'offline'` events were conflating network connectivity with PN connection status, so I've renamed their `category` properties to reflect that. There are certain scenarios where there may be network connectivity but no requests can be successfully made (for example if the user is on a VPN that has yet to establish a connection), and these weren't being handled. When there's no network connectivity and a subscribe request is aborted, the error passed by superagent is structured like this: 

```
{
    "crossDomain":true,
    "message": "Request has been terminated…"
    "method":"GET",
    "stack": (…),
    "url":"https://ps16.pubnub.com/v2/subscribe/MySubKey/channelId",
    "status": undefined
}
```

However, the only check performed in the `_detectErrorCategory` method of `networking.js` to determine if it's a network error was looking for `error.code === 'ENOTFOUND'`, so these errors were getting categorized as `PNUnknownCategory` and the polling would never get kicked off (this is reproducible by firing up a client and throttling network requests). Finally, this PR adds a `PNReconnectedCategory` status category and announces it when the `ReconnectionManager` executes `onReconnect`.

### CODE REVIEW

**Comment by MaxPresman** [comment]
@CliftonH thanks for the PR~, i'll give it a 👀 over the day

File: src/core/components/networking.js [hunk 1]
```diff
@@ -121,6 +121,7 @@ export default class {
 
   _detectErrorCategory(err: Object): string {
     if (err.code === 'ENOTFOUND') return 'PNNetworkIssuesCategory';
+    if (err.status === 0 || (err.hasOwnProperty('status') && typeof err.status === 'undefined')) return 'PNNetworkIssuesCategory';
```
Comment (MaxPresman) [inline src/core/components/networking.js]: 👍 👍

**Comment by MaxPresman** [comment]
this is awesome! will cut it over into a release in the morning

**Comment by MaxPresman** [comment]
`4.0.6` is on npm : )

**Comment by supertinou** [comment]
Thanks @CliftonH!

**Comment by CliftonH** [comment]
My pleasure, @supertinou! And thanks for the quick review/merge @MaxPresman.

**Pull request merged**